### ISSUE TITLE
[DATA] Sentiment analysis - Create script using PySpark with Logistic Regression model

### ISSUE THREAD

**Issue by ZDev-19** [unmsmfisi-socialapplication/social_app#265]

**Comment by syntx8** [comment]
# Machine Learning Model Processing and Training Script
Author: @syntx8 
Creation Date: [03/10/2023]

- Fulfilling the task assigned in #254, I proceed to code.

## Description:
This script is used to process data from the file, train a Spark-based Machine Learning model, and evaluate its performance. The trained model can be subsequently used for making predictions.

**Functions:**
- read_file(fileUrl, spark): Reads the file and load it into a Spark DataFrame.
- pre_process(df): Performs data preprocessing, including text cleaning, tokenization, stop words removal, and stemming.
- train_test_split(df): Splits the dataset into training and testing sets.
- details_table(train_predictions, test_predictions): Displays summary tables of predictions for the training and testing sets.
- evaluate_model(predictions, labelCol="label", predictionCol="prediction"): Evaluates the model's performance using metrics such as area under the ROC curve, F1 score, and accuracy.
- logistic_regression(train_data, test_data): Trains a logistic regression model using cross-validation and returns a summary of performance on the training and testing sets.

**Usage:**
1. Provide the path to the file containing the data in the 'url' variable.
2. Run the script to load, preprocess, and train the model.
3. The model's performance is evaluated, and metrics are displayed.

**Reason to Keep Model Evaluation**
Model evaluation is important for several reasons:
- It allows measuring the quality and accuracy of the trained model.
- It provides insights into how the model generalizes to new data.
- It helps fine-tune hyperparameters and improve the model.
- It is essential for informing stakeholders about the model's performance.
Therefore, keeping model evaluation is crucial to ensure code quality and usefulness.

### Main code
I show the main code where the PATH will be placed where the future database assigned to Data by DevOps will be found.

```
if __name__ == "__main__":
    spark = SparkSession.builder.appName("SocialApp").getOrCreate()
    url = "PATH"  # Provide the path to the file
    df = read_file(url, spark)
    df = pre_process(df)
    train_data, test_data = train_test_split(df)
    train_summary, test_summary = logistic_regression(train_data, test_data)
    spark.stop()
```
